bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,222 @@
+                    settings.getFeatures().getBluetooth().toString());
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_ENABLED, pmEnabled.toString());
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_TEMP_CHANNELS, String.valueOf(tempChannels));
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_CHANNELS, String.valueOf(pmChannels));
+            updateProperties(properties);
+
+            // Update channel state
+            for (Channel channel : thing.getChannels()) {
+                State state = wlanThermoEsp32CommandHandler.getState(channel.getUID(), data, settings);
+                if (state != null) {
+                    updateState(channel.getUID(), state);
+                } else {
+                    // if we could not obtain a state, try trigger instead
+                    String trigger = wlanThermoEsp32CommandHandler.getTrigger(channel.getUID(), data);
+                    if (trigger != null) {
+                        triggerChannel(channel.getUID(), trigger);
+                    }
+                }
+            }
+        } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Update failed: " + e.getMessage());
+            ScheduledFuture<?> oldScheduler = pollingScheduler;
+            if (oldScheduler != null) {
+                oldScheduler.cancel(false);
+            }
+            for (Channel channel : thing.getChannels()) {
+                updateState(channel.getUID(), UnDefType.UNDEF);
+            }
+            checkConnection();
+        }
+    }
+
+    private void push() {
+        if (data == null) {
+            return;
+        }
+        data.getChannel().forEach(c -> {
+            try {
+                String json = gson.toJson(c);
+                URI uri = config.getUri("/setchannels");
+                int status = httpClient.POST(uri).content(new StringContentProvider(json), "application/json")
+                        .timeout(5, TimeUnit.SECONDS).send().getStatus();
+                if (status == 401) {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No or wrong login credentials provided. Please configure username/password for write access to WlanThermo!");
+                } else if (status != 200) {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Failed to update channel "
+                            + c.getName() + " on device, Statuscode " + status + " on URI " + uri.toString());
+                    logger.debug("Payload sent: {}", json);
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (InterruptedException | TimeoutException | ExecutionException | URISyntaxException e) {
```

Comment:
An you exit the push method as soon as possible if you catch an InterruptedException since that means that openhab is trying to shutdown. You should not delay shutdown if at all possible.